### DISCUSSION TITLE
Render geopackage in uno platform

### DISCUSSION THREAD

**Discussion by electriceop3** [unoplatform/uno#12319] [Q&A] [answered]
Does anybody know a good way to display a geopackage in uno platform? I have heard of arcgis, but when I tried to use it, it threw an error:  The 'Esri.ArcGISRuntime' nuget package cannot be used to target 'net7.0' for output type 'Exe'. Only platform-agnostic class libraries or Windows, iOS and Android platforms are supported. I have also heard of sharpmap, but I don't know how to get it to work.

**Comment by dotMorten** [comment, marked as answer]
Hi! Devlead on the ArcGISRuntime stuff here. To build an app with the ArcGIS Maps SDK, you will need to add a platform-identifier, like `net7.0-windows10.0.19041.0`, `net7.0-ios`, `net7.0-android` or `net7.0-maccatalyst` (ie only the platforms supported by it). You can only target the `net7.0` base for class libraries.
Having said that, there is no UI component for Uno, so you'll only be able to use the business-logic bits of the package, or create a Uno wrapper for the WPF/UWP/WinUI/iOS/Android UI packages if you want to render a map on-screen.

**Reply by electriceop3** [reply]
Okay, thanks! This is for a school project, and I was planning on hosting it on the web via wasm so that my classmates and teachers would easily be able to access it. How would I create a uno wrapper, would it work for wasm, or would it be easier to look for a different package that can work for wasm? If I should look for something else, do you have any recommendations?

**Reply by dotMorten** [reply]
No wasm isn’t a platform that is supported. You’d have to resort to use the JavaScript API instead.

**Reply by electriceop3** [reply]
Okay. I haven't learned javascript (I plan to after I build a few projects in c#), so it would be easier to use c#. I found this: https://www.esri.com/en-us/arcgis-marketplace/listing/products/0fa9523e11bc4b1baa164afb007c4715 and was wondering if it is worth looking into at all. If not, I'll figure it out. Thank you for your answers!

**Reply by dotMorten** [reply]
yup that’s an option too. I know the people working on it and they know what they’re doing (one of them actually used to work on the ArcGIS maps Sdk for .NET)

**Reply by electriceop3** [reply]
Cool, sounds good, thanks!

**Reply by jeromelaban** [reply]
There's also a tutorial on how to [embed a Javascript component in an Uno app](https://platform.uno/docs/articles/interop/wasm-javascript-1.html). You can this way wrap any existing javascript-based content in a XAML UI Element.

**Reply by electriceop3** [reply]
Very cool, thanks! For future projects: Does this work for other platforms? I know that .net maui blazor has some ability to work with javascript.

**Reply by jeromelaban** [reply]
@electriceop3 this particular javascript embedding technique does not exist for iOS/Android/Catalyst, but you can [embed native controls](https://platform.uno/docs/articles/platform-specific-csharp.html?q=native#a-simple-example). For Skia targets, an upcoming feature in 4.9 will allow for overlaying native controls on top of our Skia renderer.

**Reply by electriceop3** [reply]
Cool, thanks!
